### PR TITLE
[web-app] page load improvements

### LINKED ISSUE #1201: [performance] Reduce initial web-app page load size
Here's a sample of the initial load sizes and chunks:

```
├ ○ /mdx/tab-three                                                        777 B          3.42 MB
├ ○ /mdx/tab-two                                                          780 B          3.42 MB
├ ○ /mdx/table                                                            993 B          3.42 MB
├ ○ /mdx/tabs                                                             875 B          3.42 MB
├ ○ /mdx/title                                                            597 B          3.42 MB
├ ○ /mdx/type                                                             641 B          3.42 MB
├ ○ /mdx/typesetstyle                                                     653 B          3.42 MB
├ ○ /mdx/video/video                                                      1.21 kB        3.42 MB
└ ○ /standards                                                            563 B          3.42 MB
+ First Load JS shared by all                                             3.48 MB
  ├ chunks/framework-89edbea092cc1347.js                                  42.5 kB
  ├ chunks/main-e824ab101d61bc6a.js                                       35.8 kB
  ├ chunks/pages/_app-48b02d513e3a046b.js                                 3.3 MB
  ├ chunks/webpack-db21431a018fe470.js                                    42.2 kB
  └ css/a1814c3397b7e003.css                                              65.6 kB
```

Pretty much every page that is generated requires a minimum of ~3.4 MB for its load. This chunk causes delays in rendering the side nav and also slows down the first contentful paint.

This issue is to investigate reducing the size of this initial load to increase page load performance across the board.

### PR DESCRIPTION
**Pull request by jharvey10**
- Improve page load times by using dynamic imports for large chunks (pictograms and icons metadata)
- Fix some typos
- Remove library provider and context in favor of a prop
- Remove IDL-site conditional from icons action bar

Fixes #1201

#### Testing / reviewing

- Ensure feature parity between old icons and pictograms pages and new ones
- Run a production build of the web-app `npm -w services/web-app run build`, run the production build locally `npm -w services/web-app run start`, and ensure all pages load faster than before (depending on network speed)

### CODE REVIEW

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Looks like we lost the correct padding at the top of the page when switching from mdx to js page. Wondering if we should move this style to a global wrapper, although may add incorrect spacing to other pages if we do that. Can probably look at that in a different issue unless its an easy fix.

Screen Shot 2022-09-01 at 9 56 19 AM

Screen Shot 2022-09-01 at 9 58 20 AM


Everything else is running / working great!

**Comment by jdharvey-ibm** [comment]
@alisonjoseph Fixed! Will upload it shortly. For now, since there was already a class applying `spacing-07`, I just changed it to `spacing-11`, but we can definitely move it to a top-level style. I'd just probably need help figuring out all the places to remove existing padding in favor of that one. (Not sure if it's just MDX pages or if there are other ones that would benefit from that as well)

**Comment by alisonjoseph** [comment]
@jdharvey-ibm yea we should probably make a separate issue to look into this, I think its going to be a bit of a game of whack-a-mole, and syncing with design to make sure we have consistent top spacing in the correct places.

File: services/web-app/pages/elements/icons/library.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    <>
+      <NextSeo {...seo} />
+      <PageHeader title={seo.title} withTabs={true} />
+      <PageTabs tabs={tabs} />
+      <IconLibrary iconCategoryMetadata={iconCategoryMetadata} iconMetaData={iconMetadata} />
+    </>
+  )
+}
+
+export const getStaticProps = async () => {
```
Comment (francinelucca) [inline services/web-app/pages/elements/icons/library.js:60]: you could use getStaticProps inside the .mdx file (see services/web-app/pages/designing/get-started.mdx), is there a reason you chose to convert to .js file instead? (just curious here)

**Review by francinelucca — CHANGES_REQUESTED** [review]
UI and build LGTM
PictogramLibrary and IconLibrary storybook no longer working for me

**Comment by francinelucca** [comment]
forgot to attach pic 
image

**Review by alisonjoseph — APPROVED** [review]

**Review by francinelucca — APPROVED** [review]
LGTM

**Pull request merged**